src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -292,19 +304,18 @@ uct_dc_mlx5_ep_rand_arb_group(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 static UCS_F_ALWAYS_INLINE ucs_arbiter_group_t*
 uct_dc_mlx5_ep_arb_group(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 {
-    return (uct_dc_mlx5_iface_is_policy_shared(iface)) ?
-            uct_dc_mlx5_ep_rand_arb_group(iface, ep) : &ep->arb_group;
+    return uct_dc_mlx5_iface_is_policy_shared(iface) ?
+                   uct_dc_mlx5_ep_rand_arb_group(iface, ep) :
+                   &ep->arb_group;
 }
 
 static UCS_F_ALWAYS_INLINE void
 uct_dc_mlx5_iface_dci_sched_tx(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 {
-    if (uct_dc_mlx5_iface_is_policy_shared(iface)) {
-        ucs_arbiter_group_schedule(uct_dc_mlx5_iface_tx_waitq(iface),
-                                   uct_dc_mlx5_ep_rand_arb_group(iface, ep));
-    } else if (uct_dc_mlx5_iface_dci_has_tx_resources(iface, ep->dci)) {
+    if (uct_dc_mlx5_iface_dci_has_tx_resources(iface, ep->dci) ||
+        uct_dc_mlx5_is_dci_shared(iface, ep->dci)) {
```

Comment:
Seems like both these functions access the dci array by ep->dci index to check something.
Maybe worth uniting the checks by getting pointer to dci object first?
BTW, similar check exists in uct_dc_mlx5_iface_dci_get